.github/workflows/flaky-tests.yml

```diff
@@ -24,9 +24,11 @@ jobs:
                   node-version-file: '.nvmrc'
                   cache: npm
 
-            - name: Npm install
+            - name: Npm install and build
+              # We don't have to build the entire project, just the action itself.
```

Comment:
It's not clear why `npm run build:package-types` does what this comment says which is "build the action itself". Maybe say something like "We don't have to build the entire project, just the action itself, which uses TypeScript as its compiler."

Reply:
Or we move types to their own file and import them with JSDoc to avoid all the hassle.

Reply:
I also think TypeScript is a good fit here as it adds safety to something that's error-prone but difficult to test on your local machine.